### DISCUSSION TITLE
Purpose to improve API

### DISCUSSION THREAD

**Discussion by mguyard** [iopool/community#4] [Ideas]
Hello,

I create this topic to start to centralize some feature request in API.

### Type

API don't include type (pool or spa). This can be an interesting information

### Treatment in progress

API include if action is required. It's a boolean. But we don't have information if treatment is in progress.

### Indicator color (green/orange/red) for ORP, pH and temperature

Like we have in app, it can be interesting to receive indicator color. My plugin using API try to generate color based on value like we can see here for each values.

![IMG_646BF6B9985C-1](https://user-images.githubusercontent.com/5262395/132088664-59cc9331-f327-4130-a332-d45b07a331bf.jpeg)

But if we can receive it directly thru API it can be better.

### Flag color (green/orange/red)

Like we have in app, it can be interesting to receive flag color. My plugin using API try to generate color flag based on color (see above) of each values (orp,pH, temperature). If one flag is red, the flag is red. If one is orange = flag is orange else it's green.
I don't know if it's the right way but if this information is shared by API it's will be similar to app.

### Actions

Have a array with all actions recommanded and if we have product in stock to apply it, like this we can have a automation notification when action is required but no product in stock for this action,, can permit user to purchase product quickly before open app and discover it.

### API purpose evolution

```
[{
        id: string;
        title: string;
        type: "POOL" | "SPA"; # NEW
        mode: "STANDARD" | "OPENING" | "WINTER" | "INITIALIZATION";
        hasAnActionRequired: boolean;
        hasTreatmentInProgress: boolean;
        latestMeasure: {
            flag_status: "GREEN" | "ORANGE" | "RED"; # NEW
            temperature: number;
            temperature_status: "GREEN" | "ORANGE" | "RED"; # NEW
            ph: number;
            ph_status: "GREEN" | "ORANGE" | "RED"; # NEW
            orp: number;
            orp_status: "GREEN" | "ORANGE" | "RED"; # NEW
            mode: "standard" | "live" | "maintenance" | "manual" | "backup" | "gateway";
            isValid: boolean;
            measuredAt: string;
        } | null;
        advice: {
            filtrationDuration: number | null;
        };
        actions: {
             [{
                   title: string; #NEW : Title of action like we have in app
                   details: string; #NEW: Title of action like we have in app
                   measureTrigger: "PH" | "ORP"; #NEW: Action is required based on a bad value on which measure (orp
                   validStock: boolean; #NEW: If stock permit to apply this action
              }]
        }
}]
```

**Comment by Septain21** [comment]
Very good idea, thank you

**Comment by mguyard** [comment]
Hi @mehdyouras 

Do you think part or all of these evolution can be integer ?
It would be very useful for automation.

Thanks

**Comment by oserani** [comment]
Sorry for the silly question - before I start developing interfaces (including shortcuts for Siri), I try to model/understand the api but submitting an https:// request to the API through a browser.  I have not been able to get a valid response (I either get "forbidden" or "missing api key") although I am including it in the https:// in the browser.  Could somebody let me know what that syntax looks like?  Thanks!

**Reply by mguyard** [reply]
This is not the right place for this question (better here : https://github.com/iopool/community/discussions/1) but this is example from curl :
First you need to request all pool to get the pool id : 
`curl --location 'https://api.iopool.com/v1/pools/' --header 'x-api-key: <hereyourapi>'`
In the reply, you will have the pool id for each pool.

To request pool information you need to call :
`curl --location 'https://api.iopool.com/v1/pool/<hereyourpoolid>' --header 'x-api-key: <hereyourapi>'`

Response format is the same as explain here : https://help.iopool.com/fr/articles/5537423-api-publique-iopool

**Comment by oserani** [comment]
Thanks Marc for the guidance.  I guess my question is even simpler than that.  How do I strictire that cURL command in a straight http rquest in a browser such as safari so I can automate it into an iphone shortcut.

Thanks again!

**Reply by mguyard** [reply]
You can't as you need to push api key in http header.
